src/rules/page/required-property-values-rule.js

```diff
@@ -142,6 +164,18 @@ const RequiredPropertyValuesRule = class extends RpdeRule {
         ) {
           invalidProps.kind += 1;
         }
+        if (
+          typeof item.id !== 'undefined'
+          && typeof item.id !== 'string' && typeof item.id !== 'number'
+        ) {
+          invalidProps.id += 1;
+        }
+        if (
+          typeof item.data !== 'undefined'
+          && (typeof item.data !== 'object' || item.data instanceof Array)
```

Comment:
Not sure if this is caught by other rules, but this allows for `null`, which has type `object`. It could in theory also have other clashes like with `Date`s, etc, though these are not possible at present as the data comes from JSON parsing.

I would suggest using lodash's [`isPlainObject`](https://lodash.com/docs/#isPlainObject) function here to be more clear and to be more ironclad

Reply:
Good catch @lukehesluke will push a new update that includes this